src/main/grammars/RustParser.bnf

```diff
@@ -800,31 +801,31 @@ private TypeAscription ::= ':' TypeReference
 private TypeReferenceInner ::= ArrayType
                                | RefLikeType
                                | FnPointerType
-                               | MacroType
                                // External rule to find the difference between `(T)` and `(T,)`
                                | <<tupleOrParenType TypeReferenceInner TupleType_upper>>
                                | ExplicitTraitType
                                | TrivialBaseType
                                // External rule to find the difference between `T` and `T+T`
                                | <<baseOrTraitType PathGenericArgsWithoutColons ImplicitTraitType TraitType_upper>>
+                               | MacroType
```

Comment:
Re-ordered. `MacroType` and `BasePath` both start with a path, so this is a double-parsing, but `MacroType` is a very rare case. `baseOrTraitType` is updated to reject `MacroType`.